### PR TITLE
BREAKING CHANGE: Move all regexes to rules

### PR DESCRIPTION
**Pull request by UziTech**
**Marked version:** 14.1.3

## Description

Move all regular expressions to rules.ts to be able to reuse similar regular expressions and spped up marked by creating the regular expressions once instead of inside a loop like the tokenizer or parser loops. These rules will be in an object exported as `other` by rules.ts. 

## Contributor

- [x] Test(s) exist to ensure functionality and minimize regression (if no tests added, list tests covering this PR); or,
- [ ] no tests required for this PR.
- [ ] If submitting new feature, it has been documented in the appropriate places.

## Committer

In most cases, this should be a different person than the contributor.

- [ ] CI is green (no forced merge required).
- [ ] Squash and Merge PR following [conventional commit guidelines](https://www.conventionalcommits.org/).

### CODE REVIEW

**Comment by UziTech** [comment]
@race-of-sloths

**Comment by race-of-sloths** [comment]
@UziTech Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
New Sloth joined the Race! Welcome!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://race-of-sloths.com/images/bot-banner-dark.svg">
    <source media="(prefers-color-scheme: light)" srcset="https://race-of-sloths.com/images/bot-banner-light.svg">
    Shows inviting banner with latest news.
</picture>](https://race-of-sloths.com/latest-news)

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/UziTech?type=bot&pr=markedjs/marked/3519&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/UziTech?type=bot&pr=markedjs/marked/3519&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/UziTech)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command. Alternatively, autoscoring [1,2] will be applied for this pull request

</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/UziTech)

For maintainers:
- Score pull requests that participate in the Race of Sloths and receive a reward
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Invite contributor** `@race-of-sloths invite` to invite the contributor to participate in a race or include it, if it's already a runner.
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

File: src/rules.ts
Comment (styfle) [inline src/rules.ts:3]:
```suggestion
export function edit(regex: string | RegExp, opt = '') {
```

Can you set the default here and remove line 5?

File: src/rules.ts
Comment (styfle) [inline src/rules.ts:22]: Should this nee like be removed so the JsDoc shows up on the const?

Comment (styfle) [reply src/rules.ts:22]: Or better yet, remove the comment since it’s self-explanatory

**Review by styfle — APPROVED** [review]

**Comment by race-of-sloths** [comment]
🥁  __Score it!__

@styfle, please score the PR with `@race-of-sloths score [1/2/3/5/8/13]`. The contributor deserves it.
If no scoring is provided within 24 hours, this PR will be scored as 2 🦥

**Pull request merged**
